### PR TITLE
Make Cypress Tests Environment Agnostic

### LINKED ISSUE #1263: Make cypress tests run on multiple environments
**Is your feature request related to a problem? Please describe.**
Now that we have a live environment we need our cypress tests to run on different environments with different assets 

**Describe the solution you'd like**
Following https://github.com/bbc/simorgh/pull/1236

A PR in Simorgh to split out our tests into local/test/live equivalents, either through folders or config file. The aim being to run Test cypress tests on Live, against Live URLs, without duplication of effort.

In tandem with a PR to Simorgh-Infrastructure to execute the right scripts defined by the above PR (issue: https://github.com/bbc/simorgh-infrastructure/issues/245).

**Testing notes**
Will sit down with a developer and run through the best approach to make the tests environment agnostic. We will sat down at a later date to discuss which tests should/should not be included for each environment.

**Dev insight:** 
Not writing any new tests - at the end of this work, our Test tests should be running on Live, but with live URLs.

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by jamesbrumpton**
This makes the Cypress tests environment agnostic so that we can run them against all environments.

Resolves #1263 

**Overall change:** We've made the Cypress tests environment agnostic but also pointed the `LIVE` `BASE_URL` to `Mozart_api`.

**Code changes:**

- _Make Cypress tests agnostic._
- _Point `LIVE BASE_URL` to `Mozart_api`._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
-	"video": false,
-	"baseUrl": "http://localhost:7080"
-}
+  "video": false,
+  "env": {
+    "APP_ENV": "local"
+  }
+}
```
Comment (dr3) [inline cypress.json]:
```suggestion
}

```
Just needs an EOF blank line

File: .env [hunk 2]
```diff
@@ -1,4 +1,6 @@
-# On running the 'build' commands this file is replaced by the files in `env-config/`
-# To ignore and keep this file we have a command in `postshrinkwrap` which assumes this file is not changing
-# The command is `git update-index --assume-unchanged .env` 
-# Source: https://wildlyinaccurate.com/git-ignore-changes-in-already-tracked-files/
+SIMORGH_BASE_URL=http://localhost:7080
```
Comment (dr3) [inline .env]: This file shouldn't have been changed, not sure @pjlee11's trick worked. Can be reset with `git checkout latest .env`

**Review by dr3 — CHANGES_REQUESTED** [review]

File: .env [hunk 3]
```diff
@@ -2,3 +2,4 @@
 # To ignore and keep this file we have a command in `postshrinkwrap` which assumes this file is not changing
 # The command is `git update-index --assume-unchanged .env` 
 # Source: https://wildlyinaccurate.com/git-ignore-changes-in-already-tracked-files/
+SIMORGH_BASE_URL=http://localhost:7080
```
Comment (dr3) [inline .env]:
```suggestion
```

**Review by dr3 — COMMENTED** [review]
👍 LGTM, wont officially approve though as I paired :)

This will fail on Live however as Live Asset `c9rpqy7pmypo` doesn't have an inline link. Jim is looking at fixing that

**Review by pjlee11 — COMMENTED** [review]
Looks sensible to me. Is `https://www.mozart-routing.api.bbci.co.uk` publicly available? I'm trying to think if there are any cases where `SIMORGH_BASE_URL` might fail if there is a cert/exception to be confirmed at mozart routing

**Comment by dr3** [comment]
@pjlee11 It is public, this is needed purely untill the STM change

**Comment by pjlee11** [comment]
I think with the fact we have to confirm the HTTPS exception (see screenshot) the Cypress tests may fail to get data when using that `SIMORGH_BASE_URL`. 

TL;DR - I'm 90% sure this will fail the LIVE E2Es in our deployment

image

**Comment by dr3** [comment]
This is now blocked on https://github.com/bbc/simorgh/pull/1294

File: cypress/integration/footerSpec.js [hunk 4]
```diff
@@ -10,7 +11,7 @@ describe('Footer Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
     // Only 'c9rpqy7pmypo' & 'c85pqyj5m2ko' are available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/footerSpec.js]: You can remove this comment from here. 
```suggestion
```

File: cypress/integration/headerSpec.js [hunk 5]
```diff
@@ -7,7 +8,7 @@ import {
 describe('Header Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
-    cy.visit('/news/articles/c9rpqy7pmypo');
+    cy.visit(`/news/articles/${config.assets.newsThreeSubheadlines}`);
     // Only 'c9rpqy7pmypo' & 'c85pqyj5m2ko' are available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/headerSpec.js]: You can remove this
```suggestion
```

File: cypress/integration/metaNewsSpec.js [hunk 6]
```diff
@@ -10,7 +11,7 @@ describe('Article Meta Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
     // Only 'c9rpqy7pmypo' & 'c85pqyj5m2ko' are available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/metaNewsSpec.js]:
```suggestion
```

File: cypress/integration/metaPersianSpec.js [hunk 7]
```diff
@@ -9,7 +10,7 @@ describe('Persian Article Meta Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
     // Only 'c9rpqy7pmypo' & 'c85pqyj5m2ko' are available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/metaPersianSpec.js]:
```suggestion
```

File: cypress/integration/newsBodySpec.js [hunk 8]
```diff
@@ -17,7 +18,7 @@ describe('Article Body Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
     // Only 'c9rpqy7pmypo' & 'c85pqyj5m2ko' are available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/newsBodySpec.js]:
```suggestion
```

File: cypress/integration/persianBodySpec.js [hunk 9]
```diff
@@ -13,7 +14,7 @@ describe('Article Body Tests', () => {
   // eslint-disable-next-line no-undef
   before(() => {
     // Only 'cwv2xv848j5o' is available within the PROD enviroment
```
Comment (sareh) [inline cypress/integration/persianBodySpec.js]:
```suggestion
```

File: cypress/integration/newsBodySpec.js [hunk 10]
```diff
@@ -86,7 +87,7 @@ describe('Article Body Tests', () => {
     );
   });
 
-  it('should have a working first inline link', () => {
-    clickInlineLinkAndTestPageHasHTML('main a', '/news/articles/c85pqyj5m2ko');
-  });
+  // it('should have a working first inline link', () => {
```
Comment (sareh) [inline cypress/integration/newsBodySpec.js:89]: Instead of commenting out this test, I'd suggest changing the assertion, so we continue to test the first inline link of a page.

**Review by sareh — CHANGES_REQUESTED** [review]
Looks great so far. There are some comments to tidy the tests up.

**Comment by bcmn** [comment]
Currently the metadata Cypress tests have hard-coded values for the OpenGraph url tag.

[In `metaNewsSpec`](https://github.com/BBC-News/simorgh/blob/6dd1bf4ffb4b627664da9537abac644c07090032/cypress/integration/metaNewsSpec.js#L83).

[In `metaPersianSpec`](https://github.com/BBC-News/simorgh/blob/6dd1bf4ffb4b627664da9537abac644c07090032/cypress/integration/metaPersianSpec.js#L39).

This means the Cypress tests fail when you run `npm run test:e2e:live_env`, with a failure like:
```
Timed out retrying: expected '<meta>' to have attribute 'content' with the value 'https://www.bbc.com/persian/articles/cwv2xv848j5o', but the value was 'https://www.bbc.com/persian/articles/c5ml5n7w4m3o'
```
This didn't show on CI because we're not running any of the new commands there yet.

I think we'd want to replace it with something like:
```
openGraphMeta(
    'شاید خیلی طول نکشد که زمانی برسد که وقتی خسته هستید و مثلا هوس فنجان قهوه‌ای را کردید، پهپادی را ببینید که با قهوه سراغتان می‌آید.',
    'https://news.files.bbci.co.uk/ws/img/logos/og/persian.png',
    'BBC News فارسی',
    'fa',
    'BBC News فارسی',
    'پهپادی که برایتان قهوه می‌آورد',
    'article',
    `https://www.bbc.com/persian/articles/${config.assets.persian}`,
  );
```

**Review by sareh — APPROVED** [review]

**Review by bcmn — APPROVED** [review]

**Comment by bcmn** [comment]
Moving to "ready for merge" as tested by Sareh, James & I.

**Review by dr3 — APPROVED** [review]

**Comment by ChrisBAshton** [comment]
Merged despite Jenkins fail, because Jenkins was consistently passing tests etc, this looks like an internal workspace-clearing issue. Investigating separately.
